src/microservices/Updating/Microservices.UpdateValues/Execution/UpdateValuesHost.cs

```diff
@@ -0,0 +1,29 @@
+﻿using JetBrains.Annotations;
+using Rdmp.Core.Repositories;
+using Smi.Common;
+using Smi.Common.Execution;
+using Smi.Common.Options;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microservices.UpdateValues.Execution
+{
+    public class UpdateValuesHost : MicroserviceHost
+    {
+        UpdateValuesQueueConsumer Consumer {get;set;}
+
+        public UpdateValuesHost([NotNull] GlobalOptions globals, IRabbitMqAdapter rabbitMqAdapter = null, bool loadSmiLogConfig = true, bool threaded = false) : base(globals, rabbitMqAdapter, loadSmiLogConfig, threaded)
+        {
+        }
+
+        public override void Start()
+        {
+            
+            IRDMPPlatformRepositoryServiceLocator repositoryLocator = Globals.RDMPOptions.GetRepositoryProvider();
+            Consumer = new UpdateValuesQueueConsumer(Globals.UpdateValuesOptions,repositoryLocator.CatalogueRepository);
```

Comment:
Would this not be better located in the constructor?

Reply:
I don't think it makes a big difference, the `MicroserviceHostBootstrapper` pattern at least used to expose errors more cleanly in the `Start` method.  That's what `CohortExtractorHost` does and `DicomRelationalMapperHost`.  If a constructor blows up your object can't be used but if it fails to start it can at least report itself to fatal error queues etc.  To me once you have a Start method then its basically the same as the constructor except that you can try it again or log it better if it goes wrong without having to have an outside of class perspective.